### PR TITLE
Feature/task 3

### PR DESCRIPTION
**Pull request by medvedovic**
Done-ish

### CODE REVIEW

File: onboarding-task/src/components/List.jsx [hunk 1]
```diff
@@ -1,69 +1,30 @@
...
+            </li>
+          ))
+        }
+      </ol>
+      <ListItemInput onCreateItem={onCreateItem} />
+    </div>
+  </div>
+);
+
+List.propTypes = {
```
Comment (Suzii) [inline onboarding-task/src/components/List.jsx]: Missing `displayName`.

File: onboarding-task/src/reducers/items/data/item.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
-import { UPDATE_ITEM } from '../constants/actionTypes';
-import { ListItemData } from '../models/ListItemData';
+import { UPDATE_ITEM } from '../../../constants/actionTypes';
+import { ListItemData } from '../../../models/ListItemData';
 
-export const updateItemReducer = (state = {}, action) => {
+export const item = (state = ListItemData(), action) => {
```
Comment (PetrSvirak) [inline onboarding-task/src/reducers/items/data/item.js]: Use `new` here – see [docs](https://facebook.github.io/immutable-js/docs/#/Record)

File: onboarding-task/test/reducers/items/data/data.test.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    [_id2, new ListItemModel({ id: _id2, value: 'Master React' })],
+  ]);
+
+  it('creates new item properly', () => {
+    const expectedStoreState = new Map([
+      [_id1, new ListItemModel({ id: _id1, value: 'Make coffee' })],
+      [_id2, new ListItemModel({ id: _id2, value: 'Master React' })],
+      [_id3, new ListItemModel({ id: _id3, value: 'Make a sandwich' })],
+    ]);
+    const dummyIdGenerator = () => new ListItemModel({ id: _id3, value: 'Make a sandwich' });
```
Comment (PetrSvirak) [inline onboarding-task/test/reducers/items/data/data.test.js]: Rename to `dummyItemGenerator` – it generates `item`s not `id`s :wink:

File: onboarding-task/test/utils/generateGuid.test.js [hunk 4]
```diff
@@ -3,17 +3,17 @@ import { generateGuid } from '../../src/utils/generateGuid';
 describe('generateGuid', () => {
   it('provides correct format', () => {
     const regex = /(\w{8}-\w{4}-4\w{3}-\w{4}-\w{12})/;
-    const guid = generateGuid();
+    const id = generateGuid();
 
-    const isMatch = regex.test(guid);
+    const isMatch = regex.test(id);
 
     expect(isMatch).toBe(true);
```
Comment (PetrSvirak) [inline onboarding-task/test/utils/generateGuid.test.js]: Use `toBeTruthy` and `toBeFalsy` in all occurences.

File: onboarding-task/test/actions/userActions.test.js [hunk 5]
```diff
@@ -2,46 +2,29 @@ import {
   createItem,
   updateItem,
   deleteItem,
-} from '../../src/actions/userActions';
+} from '../../src/actions/publicActions';
 import {
   CREATE_ITEM,
```
Comment (PetrSvirak) [inline onboarding-task/test/actions/userActions.test.js]: Remove unused imports.

**Pull request merged**